File: src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	
	//获取登录名
	this.getUserName=function(){
		return $http.get('../user/getUserName.do');		
	}  
	
	//用户注册
	this.add=function(entity,smscode){
		return $http.post('../user/add.do?smscode='+smscode,entity);		
	}  
	
	//发送验证码
	this.sendCode=function(phone){
		return $http.get("../user/sendCode.do?phone="+phone);
	}
	
	//获取用户信息
	this.getUserInfo=function(){
		return $http.get("../user/getUserInfo.do");
	}
	
	//获取用户信息
	this.update=function(entity){
		return $http.post("../user/update.do",entity);
	}

	//获取用户信息
	this.updatePsw=function(entity){
		return $http.post("../user/updatePsw.do",entity);
	}
		
});

File: src/main/java/com/manicure/service/UserService.java
package com.manicure.service;

import com.manicure.entity.TbUser;

public interface UserService {

	/**
	 * 用户注册
	 * @param user
	 */
	public void add(TbUser user);
	
	/**
	 * 生成短信验证码
	 * @return
	 */
	public void createSmsCode(String phone);
	
	/**
	 * 判断短信验证码是否存在
	 * @param phone
	 * @return
	 */
	public boolean  checkSmsCode(String phone,String code);
	
	/**
	 * 回显用户基本信息
	 * @param username
	 * @return
	 */
	public TbUser getByUserName(String username);
	
	/**
	 * 修改用户信息
	 * @param user
	 */
	public void update(TbUser user);


	/**
	 * 修改密码
	 * @param user
	 */
    void updatePsw(TbUser user);
}
